modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -282,6 +282,29 @@ private MapStream(Stream<Map.Entry<K, V>> underlying) {
     return underlying.map(e -> mapper.apply(e.getKey(), e.getValue()));
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Transforms the entries in the stream by applying a mapper function to each key.
+   *
+   * @param mapper  a mapper function whose return values are included in the new stream
+   * @param <R>  the type of elements in the new stream
+   * @return a stream containing the values returned from the mapper function
+   */
+  public <R> Stream<R> flatMapKeys(Function<? super K, Stream<? extends R>> mapper) {
```

Comment:
The `mapKeys()` method has 2 overloads - one taking the key and another taking both the key and value. Is there an argument for doing the same here?

Reply:
As stephen mentioned above, not sure how worthwhile these methods are since .keys().flatMap() can be used.